eng/common/scripts/Validate-Package.ps1

```diff
@@ -0,0 +1,291 @@
+        return $false
+    }
+
+    $changeLogStatus = $pkgValidationDetails.ChangeLogValidation.Status
+    $changeLogDetails  = $pkgValidationDetails.ChangeLogValidation.Message
+    $apiReviewStatus = $pkgValidationDetails.APIReviewValidation.Status
+    $apiReviewDetails = $pkgValidationDetails.APIReviewValidation.Message
+    $packageNameStatus = $pkgValidationDetails.PackageNameValidation.Status
+    $packageNameDetails = $pkgValidationDetails.PackageNameValidation.Message
+
+    $fields = @()
+    $fields += "`"PackageVersion=${versionString}`""
+    $fields += "`"ChangeLogStatus=${changeLogStatus}`""
+    $fields += "`"ChangeLogValidationDetails=${changeLogDetails}`""
+    $fields += "`"APIReviewStatus=${apiReviewStatus}`""
+    $fields += "`"APIReviewStatusDetails=${apiReviewDetails}`""
+    $fields += "`"PackageNameApprovalStatus=${packageNameStatus}`""
+    $fields += "`"PackageNameApprovalDetails=${packageNameDetails}`""
+    if ($BuildDefinition) {
+        $fields += "`"PipelineDefinition=$BuildDefinition`""
+    }
+    if ($PipelineUrl) {
+        $fields += "`"LatestPipelineRun=$PipelineUrl`""
+    }
+
+    $workItem = UpdateWorkItem -id $workItem.id -fields $fields
+    Write-Host "[$($workItem.id)]$LanguageDisplayName - $pkgName($versionMajorMinor) - Updated"
+    return $true
+}
+
+# Read package property file and identify all packages to process
+Write-Host "Processing package: $PackageName"
+$packagePropertyFile = Join-Path $ConfigFileDir "$PackageName.json"
+$pkgInfo = Get-Content $packagePropertyFile | ConvertFrom-Json
+
+$changeLogPath = $pkgInfo.ChangeLogPath
+$versionString = $pkgInfo.Version
+Write-Host "Checking if we need to create or update work item for package $packageName with version $versionString."
+$isShipped = IsVersionShipped $packageName $versionString
+if ($isShipped) {
+    Write-Host "Package work item already exists for version [$versionString] that is marked as shipped. Skipping the update of package work item."
+    exit 1
+}
+
+Write-Host "Validating package $packageName with version $versionString."
+
+# Change log validation
+$changelogStatus = ValidateChangeLog $changeLogPath $versionString
+
+# API review and package name validation
+$apireviewDetails = VerifyAPIReview $PackageName $pkgInfo.Version $LanguageDisplayName
+
+$pkgValidationDetails= [PSCustomObject]@{
+    Name = $PackageName
+    Version = $pkgInfo.Version
+    ChangeLogValidation = $changelogStatus
+    APIReviewValidation = $apireviewDetails.ApiviewApproval
+    PackageNameValidation = $apireviewDetails.PackageNameApproval
+}
+
+$output = ConvertTo-Json $pkgValidationDetails
+Write-Host "Output: $($output)"
+
+# Create json token file in artifact path
+$tokenFile = Join-Path $ArtifactPath "$PackageName-Validation.json"
+$output | Out-File -FilePath $tokenFile -Encoding utf8
+
+# Create DevOps work item
+$updatedWi = CreateUpdatePackageWorkItem $pkgInfo
+
+# Update validation status in package work item
```

Comment:
Do we need to return an exit code?

Reply:
Added a check to verify status of create/update work item and removed explicit exit code from this script.